BinaryObjectScanner/Protection/CopyX.cs

```diff
@@ -114,22 +116,33 @@ public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<strin
             // Excludes files with .x64 extension to avoid flagging Professional files.
             // Sorts list of files in ZDAT* so just the first file gets pulled, later ones have a chance of the ring 
             // intersecting the start of the file.
-            var fileList = files.Where(f => !f.EndsWith(".x64", StringComparison.OrdinalIgnoreCase))
-                .Where(f =>
+
+            string[] dirs = ["ZDAT", "System/", "System\\"];//Kenny's Adventure uses System instead of ZDAT.
+            List<string>? lightFiles = null;
+            // TODO: Compensate for the check being run a directory or more higher
+            var fileList = files.Where(f => !f.EndsWith(".x64", StringComparison.OrdinalIgnoreCase));
+            {
+                foreach (var dir in dirs)
                 {
-                    // TODO: Compensate for the check being run a directory or more higher
-                    f = f.Remove(0, path.Length);
-                    f = f.TrimStart('/', '\\');
-                    return f.StartsWith("ZDAT", StringComparison.OrdinalIgnoreCase);
-                })
-                .OrderBy(f => f)
-                .ToList();
-
-            if (fileList.Count > 0)
+                    lightFiles = fileList.Where(f =>
+                        {
+                            f = f.Remove(0, path.Length);
+                            f = f.TrimStart('/', '\\');
+                            return f.StartsWith(dir, StringComparison.OrdinalIgnoreCase);
+                        })
+                        .OrderBy(f => f)
+                        .ToList();
+                    if (lightFiles.Any())
+                    {
+                        break;
+                    }
+                }
+            }
+            if (lightFiles.Count > 0)
```

Comment:
It is not guaranteed that `lightFiles` has a value at all because of how you have this setup. Either you can put all of the following code inside of the loop above to avoid needing to `break`, or you can add a secondary check for `lightFiles != null` before the `Count` check.

If you go with the former, the check would be inverted, using a `continue` instead of `break` if there were no files found.